plugin/storage/badger/factory.go

```diff
@@ -190,3 +205,56 @@ func (f *Factory) maintenance() {
 		}
 	}
 }
+
+func (f *Factory) metricsCopier() {
+	metricsTicker := time.NewTicker(f.Options.primary.MetricsUpdateInterval)
+	defer metricsTicker.Stop()
+	for {
+		select {
+		case <-f.maintenanceDone:
+			return
+		case <-metricsTicker.C:
+			expvar.Do(func(kv expvar.KeyValue) {
+				if strings.HasPrefix(kv.Key, "badger") {
+					if intVal, ok := kv.Value.(*expvar.Int); ok {
+						if g, found := f.metrics.badgerMetrics[kv.Key]; found {
+							g.Update(intVal.Value())
+						}
+					} else if mapVal, ok := kv.Value.(*expvar.Map); ok {
+						mapVal.Do(func(innerKv expvar.KeyValue) {
+							// The metrics we're interested in have only a single inner key (dynamic name)
+							// and we're only interested in its value
+							if intVal, ok := innerKv.Value.(*expvar.Int); ok {
```

Comment:
this is catching the metrics declared like this?

```go
NumLSMGets = expvar.NewMap("badger_lsm_level_gets_total")
```

Why does Badger use a map in this case? It looks like it could have more than one entry in the map, but we're collapsing all under the map's name.

Reply:
btw, if multiple values are possible, we could put that value into a tag on the metric.

Reply:
Currently it is storing in the map the directory of the LSM or Value. We have that information in the Factory properties also, but it might indeed make sense to export that information somewhere. Unless there's a place to view properties already in Jaeger?

Reply:
With the tags it would look like this:

```
# HELP jaeger_badger_lsm_size_bytes badger_lsm_size_bytes
# TYPE jaeger_badger_lsm_size_bytes gauge
jaeger_badger_lsm_size_bytes{directory="/tmp/badger043433444"} 0
```

Reply:
as long as there's a guarantee that map has only one entry, it makes more sense to go without the dir name in the label (since it might be different after restart)